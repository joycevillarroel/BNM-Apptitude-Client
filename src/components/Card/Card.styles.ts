import styled from 'styled-components'

export type IVariation = 'primary' | 'secondary'

interface ICardContainerProps {
  variation: IVariation
}

export const CardContainer = styled.div`
  width: 300px;
  border: 1px solid ${({ theme }) => theme.gray200};
`

export const SubContainer = styled.div`
  width: 300px;
  display: flex;
  padding-left: 16px;
  padding-top: 16px;
`

export const NameContainer = styled.div`
  background-color: ${({ theme }) => theme.primary};
  padding-left: 16px;
  padding-top: 4px;
  padding-bottom: 4px;
`

export const NameText = styled.text`
  background-color: ${({ theme }) => theme.primary};
  color: ${({ theme }) => theme.white};
`
export const CountContainer = styled.div`
  border-right: 3px solid ${({ theme }) => theme.black};
  padding-right: 8px;
`

export const Count = styled.text`
  color: ${({ theme }) => theme.black};
  width: 20px;
  font-size: 30px;
`
export const Percentile = styled.text<ICardContainerProps>`
  color: ${({ theme, variation }) => theme[variation]};
  font-size: 30px;
  padding-left: 16px;
`

export const IconContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  align-self: center;
  padding-left: 16px;
`
export const IconText = styled.text`
  color: ${({ theme }) => theme.gray};
  font-size: 10px;
`
export const Icon = styled.img`
  width: 16px;
`

export const PerformaceBox = styled.div`
  display: flex;
  justify-content: flex-end;
  padding: 8px;
  margin-top: 8px;
`

export const Performance = styled.text`
  color: ${({ theme }) => theme.gray200};
  font-size: 10px;
`
export const PerformanceNumber = styled.text<ICardContainerProps>`
  color: ${({ theme, variation }) => theme[variation]};
  font-size: 10px;
`
