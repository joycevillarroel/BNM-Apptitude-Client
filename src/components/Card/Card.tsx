import React from 'react'
import {
  CardContainer,
  Count,
  CountContainer,
  NameContainer,
  NameText,
  Percentile,
  Icon,
  IconContainer,
  IconText,
  SubContainer,
  Performance,
  PerformanceNumber,
  PerformaceBox,
} from './Card.styles'
import { IMetrics } from '../../redux/metrics/MetricsInterface'
import DownIcon from '../../assets/images/BNM_Metric_Down.svg'
import UpIcon from '../../assets/images/BNM_Metric_Up.svg'

interface ICard {
  metrics: IMetrics
}

export function Card({ metrics }: ICard) {
  const isPositiveVariation = metrics.week_over_week_change > 0
  const variationChange = isPositiveVariation ? 'primary' : 'secondary'
  return (
    <CardContainer data-testid='card'>
      <NameContainer>
        <NameText>{metrics.metric_name.toUpperCase()}</NameText>
      </NameContainer>
      <SubContainer>
        <CountContainer>
          <Count>
            {new Intl.NumberFormat('en-IN', {
              maximumSignificantDigits: 4,
            }).format(metrics.metric_count)}
          </Count>
        </CountContainer>
        <Percentile data-testid='icon' variation={variationChange}>
          {metrics.percentile}%
        </Percentile>
        <IconContainer>
          <Icon src={isPositiveVariation ? UpIcon : DownIcon} />
          <IconText>pctl</IconText>
        </IconContainer>
      </SubContainer>
      <PerformaceBox>
        <Performance>
          Your performace from last week{' '}
          <PerformanceNumber variation={variationChange}>
            {isPositiveVariation && '+'}{metrics.week_over_week_change}%
          </PerformanceNumber>
        </Performance>
      </PerformaceBox>
    </CardContainer>
  )
}
