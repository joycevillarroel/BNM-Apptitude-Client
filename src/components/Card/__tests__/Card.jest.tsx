import { render } from "@testing-library/react"
import { Card } from '../Card'
import { metricsGroupsMock } from "../../../redux/metrics/__tests__/MetricsMock"

describe('Card', () => {
  const metrickMock = metricsGroupsMock.metricsList[0].data[0]

  test('should render', () => {
    const { getByTestId } = render(<Card metrics={metrickMock} />)

    const component = getByTestId('card')
    expect(component).toBeTruthy()
    expect(component).toMatchSnapshot()
  })
})
