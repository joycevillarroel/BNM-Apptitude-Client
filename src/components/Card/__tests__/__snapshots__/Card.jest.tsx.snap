// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `
<div
  class="sc-beqWaB fMkGij"
  data-testid="card"
>
  <div
    class="sc-dmqHEX flHICX"
  >
    <text
      class="sc-hLseeU bRNsPK"
    >
      SESSIONS (SITE TRAFFIC)
    </text>
  </div>
  <div
    class="sc-gueYoa kEjKMf"
  >
    <div
      class="sc-eDDNvR hPERhE"
    >
      <text
        class="sc-jTrPJq hqtzJh"
      >
        1,400
      </text>
    </div>
    <text
      class="sc-gLDzan bIRRbM"
      data-testid="icon"
    >
      100.00
      %
    </text>
    <div
      class="sc-iAEyYk ihxAWM"
    >
      <img
        class="sc-qRumB chHEqg"
      />
      <text
        class="sc-fsQiph cnDdDq"
      >
        pctl
      </text>
    </div>
  </div>
  <div
    class="sc-jsMahE hgVxFc"
  >
    <text
      class="sc-kFuwaP jquKJZ"
    >
      Your performace from last week
       
      <text
        class="sc-irTswW hIVvXP"
      >
        -2
        %
      </text>
    </text>
  </div>
</div>
`;
